### PR TITLE
refactor(components, app): move MenuList folder to components directory

### PR DESCRIPTION
**Pull request by koji**
# Overview
move `MenuList` folder to components directory
most of components in MenuList to `components/atoms/MenuList`
hook.tsx is moved to `components/hooks`
add simple tests for `MenuItem`

close AUTH-629

## Risk assessment
low

### CODE REVIEW

File: app/tsconfig.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     },
     {
       "path": "../labware-library"
-    },
+    }
```
Comment (jerader) [inline app/tsconfig.json:30]: need this change?

File: components/src/atoms/MenuList/__tests__/MenuItem.test.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { COLORS } from '../../../helix-design-system'
+import { SPACING, TYPOGRAPHY } from '../../../ui-style-constants'
+
+import { MenuItem } from '../MenuItem'
+
+const render = (props: React.ComponentProps<typeof MenuItem>) => {
+  return renderWithProviders(<MenuItem {...props} />)[0]
+}
+
+describe('MenuItem', () => {
```
Comment (jerader) [inline components/src/atoms/MenuList/__tests__/MenuItem.test.tsx:15]: nice! thanks for making a test

**Review by jerader — APPROVED** [review]
lgtm!

File: app/tsconfig.json (hunk 1 above)
Comment (koji) [reply app/tsconfig.json:30]: i think this came from vscode prettier so that won't cause any issue.

**Pull request merged**